src/deepqmc/wf/base.py

```diff
@@ -6,6 +6,13 @@
 __all__ = ['state_callback']
 
 
+def init_wf(rng, hamil, ansatz, sample_size, state_callback):
+    r = hamil.init_sample(rng, sample_size)
+    params, wf_state = jax.vmap(ansatz.init, (None, 0), (None, 0))(rng, r)
+    wf_state, _ = state_callback(wf_state)
```

Comment:
Calling `state_callback` on a WF state returned by a haiku init function doesn't have any effect. This is because `ansatz.init` always returns the default initial state (something like `[0]`), regardless of what the dummy input `r` is. Since we don't need to pass an actual `wf_state` to `sampler.init`, I propose that we use this `init_wf` function to return only the initial parameters, while the initial state should always be created through the sampler initialization. See [be82647](https://github.com/ae-foster/deepqmc/commit/be8264759caaac18a7ee2ae378944cf30529634e) for an implementation of this.

Reply:
Ah, I wasn't aware of that. Thanks for clarifying.